sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -2007,6 +2007,54 @@ class DatasetSuite extends QueryTest
 
     checkAnswer(withUDF, Row(Row(1), null, null) :: Row(Row(1), null, null) :: Nil)
   }
+
+  test("SPARK-34205: Pipe Dataset") {
+    assume(TestUtils.testCommandAvailable("cat"))
+
+    val nums = spark.range(4)
+    val piped = nums.pipe("cat", (l, printFunc) => printFunc(l.toString)).toDF
```

Comment:
nit: any reason for `toDF` (as `pipe` gives a `Dataset[String]`)?

Reply:
It is just for `checkAnswer`.